static/openapi/faas-spec.json

```diff
@@ -0,0 +1,670 @@
+      "post": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Execute the latest version of a function",
+        "description": "Execute the latest version of a function for the specified function name on body.",
+        "operationId": "handle_latest_mm_execute",
+        "parameters": [
+          {
+            "name": "X-PhotonIQ-CustomerID",
+            "in": "header",
+            "description": "The customer ID",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "functionName",
+            "in": "path",
+            "description": "The function name",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "requestBody": {
+          "description": "A valid JSON body is accepted",
+          "content": {
+            "application/json": {
+              "schema": {},
+              "example": {}
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully executed the function."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/execute/{functionName}/{versionOrAlias}": {
+      "post": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Execute the function",
+        "description": "Execute the function for the specified function name on body.",
+        "operationId": "handle_mm_execute",
+        "parameters": [
+          {
+            "name": "X-PhotonIQ-CustomerID",
+            "in": "header",
+            "description": "The customer ID",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "functionName",
+            "in": "path",
+            "description": "The function name",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          },
+          {
+            "name": "versionOrAlias",
+            "in": "path",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "requestBody": {
+          "description": "A valid JSON body is accepted",
+          "content": {
+            "application/json": {
+              "schema": {},
+              "example": {}
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "Successfully executed the function."
+          }
+        }
+      }
+    },
+    "/api/faas/v1/function/{functionName}/alias/{aliasName}": {
+      "put": {
+        "tags": [
+          "Function"
+        ],
+        "summary": "Update the alias",
+        "description": "Update the alias for the specified key function name.",
```

Comment:
I don't understand.  What are you updating?

below it takes as an input: "An array of alias object that includes valid 'version' and 'weight'"

Can you please explain in this description what you are trying to accomplish when executing this API.
BTW, what is a "weight"?

Reply:
When we give an alias "dev" for a function "myFunction" for example, we need to associate it with specific versions.

For example:
`[POST] <...>/execute/myFunction/1.0.0`

Can be called by:
`[POST] <...>/execute/myFunction/dev`

But we can associate an alias for more than one version, then we need to distribute the calls according to the user needs:

For example if we have multiple versions:
`[POST] <...>/execute/myFunction/1.0.0`
`[POST] <...>/execute/myFunction/2.0.0`
`[POST] <...>/execute/myFunction/3.0.0`

```
[
  {
    "version" : "1.0.0",
    "weight": "20"
  },
  {
    "version" : "2.0.0",
    "weight": "30"
  },
  {
    "version" : "2.0.0",
    "weight": "50"
  }
]
```

The function call must be distributed according to these percentages (in a total of 100%).

This is useful if the user needs to migrate things smoothly to test if new versions are working properly.

Reply:
The whole concept of weights and distributing the calls needs to be described in the API.  Use one of the LLM to help you come up with the text to explain this.

Reply:
@dlozina-macrometa, @felipe-lopes-macrometa  -  `weight` can be from `0` to `1`. 
Please change to the next:
```
[
  {
    "version" : "1.0.0",
    "weight": "0.2"
  },
  {
    "version" : "2.0.0",
    "weight": "0.3"
  },
  {
    "version" : "2.0.0",
    "weight": "0.5"
  }
]
```